### ISSUE TITLE
Accuracy metric for binary_crossentropy

### ISSUE THREAD

**Issue by novog** [tflearn/tflearn#742]
Based on #253, #255, and #308, it seems like the Accuracy metric is supposed to work when using the binary_crossentropy loss function on one-dimensional targets. However, it seems to be ineffective, reporting the number of 1s instead of reporting the accuracy. Is there a fix for this issue?

Some code to demonstrate the problem is below. (The target function to learn is a cutoff at 0 for a specific linear combination of the inputs.) Everything works as expected when not running in binary mode. Here are the results in binary mode with 100 epochs:

    train acc: [0.49200000143051148]
    test acc:  [0.56000000238418579]
    y:         [0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0]
    pred:      [0.0, 0.98, 0.0, 0.98, 0.0, 0.99, 0.95, 0.0, 0.72, 0.99]
    y sum:     56.0

The net has clearly learned the function, but the accuracy metric does not show it. The accuracy results written out for Tensorboard are similarly incorrect, showing a constant 0.56 for "BinaryAccuracy/Validation". (Again, 56 was the number of 1s in the target output in this case.)

    def bxo_accuracy_test(is_binary, n_epochs, tensorboard_dir, run_id):
        train_size = 1000
        test_size = 100
        X = np.random.rand(train_size + test_size, 5)
        prod = X.dot([-2.0, -1.0, 0.0, 1.0, 2.0])
        np.clip(prod, 0.0, None, prod)
        class_num = np.sign(prod)
        if is_binary:
            y = class_num.reshape((class_num.shape[0], 1))
        else:
            class_num_int = np.array(class_num, dtype=np.int)
            identity = np.identity(2)
            y = identity[class_num_int]
        X_train = X[:train_size]
        X_test = X[train_size:]
        y_train = y[:train_size]
        y_test = y[train_size:]
        
        tf.reset_default_graph()
        net = tl.input_data([None, X.shape[1]])
        net = tl.batch_normalization(net)
        net = tl.relu(net)
        net = tl.fully_connected(net, 100, weights_init='xavier')
        net = tl.batch_normalization(net)
        net = tl.relu(net)
        net = tl.fully_connected(net, 50, weights_init='xavier')
        net = tl.batch_normalization(net)
        net = tl.relu(net)
        net = tl.fully_connected(net, 25, weights_init='xavier')
        net = tl.batch_normalization(net)
        net = tl.relu(net)
        final_width = 1 if is_binary else 2
        net = tl.fully_connected(net, final_width, 'sigmoid', weight_decay=0.0)
        optimizer = tl.Momentum(0.01)
        loss = tl.binary_crossentropy if is_binary else tl.categorical_crossentropy
        net = tl.regression(net, optimizer=optimizer, loss=loss, metric='accuracy')
        model = tl.DNN(net, tensorboard_dir=tensorboard_dir)
        batch_size = 100
        with tf.Session():
            model.fit(X_train, y_train, n_epochs, (X_test, y_test), True, batch_size, run_id=run_id)
            pred = model.predict(X_test)
            print('train acc: ' + str(model.evaluate(X_train, y_train)))
            print('test acc:  ' + str(model.evaluate(X_test, y_test)))
            if is_binary:
                print('y:         ' + str([round(v[0], 2) for v in y_test[:10]]))
                print('pred:      ' + str([round(v[0], 2) for v in pred[:10]]))
                print('y sum:     ' + str(y_test.sum()))
            else:
                print('y:         ' + str([str([round(w, 2) for w in v]) for v in y_test[:10]]))
                print('pred:      ' + str([str([round(w, 2) for w in v]) for v in pred[:10]]))
                print('y sum:     ' + str(y_test.sum(axis=0)))

Thanks!
